networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeMessageHandler.java

```diff
@@ -0,0 +1,299 @@
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final Spec spec;
+  private final UInt64 denebForkEpoch;
+  private final RecentChainData recentChainData;
+  private final CombinedChainDataClient combinedChainDataClient;
+  private final UInt64 maxRequestSize;
+  private final UInt64 maxBlobsPerBlock;
+  private final LabelledMetric<Counter> requestCounter;
+  private final Counter totalBlobSidecarsRequestedCounter;
+
+  public BlobSidecarsByRangeMessageHandler(
+      final Spec spec,
+      final UInt64 denebForkEpoch,
+      final MetricsSystem metricsSystem,
+      final RecentChainData recentChainData,
+      final CombinedChainDataClient combinedChainDataClient,
+      final UInt64 maxRequestSize,
+      final UInt64 maxBlobsPerBlock) {
+    this.spec = spec;
+    this.denebForkEpoch = denebForkEpoch;
+    this.recentChainData = recentChainData;
+    this.combinedChainDataClient = combinedChainDataClient;
+    this.maxRequestSize = maxRequestSize;
+    this.maxBlobsPerBlock = maxBlobsPerBlock;
+    requestCounter =
+        metricsSystem.createLabelledCounter(
+            TekuMetricCategory.NETWORK,
+            "rpc_blob_sidecars_by_range_requests_total",
+            "Total number of blob sidecars by range requests received",
+            "status");
+    totalBlobSidecarsRequestedCounter =
+        metricsSystem.createCounter(
+            TekuMetricCategory.NETWORK,
+            "rpc_blob_sidecars_by_range_requested_sidecars_total",
+            "Total number of blob sidecars requested in accepted blob sidecars by range requests from peers");
+  }
+
+  @Override
+  public Optional<RpcException> validateRequest(
+      final String protocolId, final BlobSidecarsByRangeRequestMessage request) {
+    final UInt64 finalizedEpoch = recentChainData.getFinalizedEpoch();
+    final UInt64 minEpochForBlobSidecars =
+        recentChainData
+            .getCurrentEpoch()
+            .orElse(UInt64.ZERO)
+            .minusMinZero(MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS);
+    final UInt64 minimumRequestEpoch =
+        finalizedEpoch.max(minEpochForBlobSidecars).max(denebForkEpoch);
+
+    if (!verifyRequestedEpochIsInSupportedRange(request.getStartSlot(), minimumRequestEpoch)) {
+      requestCounter.labels("resource_unavailable").inc();
+      return Optional.of(
+          new RpcException.ResourceUnavailableException(
+              "Can't request blob sidecars earlier than epoch " + minimumRequestEpoch));
+    }
+    return Optional.empty();
+  }
+
+  @Override
+  public void onIncomingMessage(
+      final String protocolId,
+      final Eth2Peer peer,
+      final BlobSidecarsByRangeRequestMessage message,
+      final ResponseCallback<BlobSidecar> callback) {
+    final UInt64 startSlot = message.getStartSlot();
+    LOG.trace(
+        "Peer {} requested {} blob sidecars starting at slot {}.",
+        peer.getId(),
+        message.getCount(),
+        startSlot);
+
+    if (!peer.wantToMakeRequest()
```

Comment:
I agree. Not sure `want` is the right verb here. I'd rename `wantToMakeRequest`  to `peerRateLimitReached` and `wantToReceiveBlobSidecars` to `blobSidecarsRequestsLimitReached`

Reply:
we can have a local private method that hides these two calls. like `doesTriggerRateLimit(callback, maxRequestSize.min(message.getCount()).longValue()))`.

Reply:
but I agree that this should be addressed in a separate PR since this is a common pattern we use in other RPC methods too. I we want to change the peer interface we can log an issue and make a consistent change later.

Reply:
Yeah, it's just a first time I see code here in gossip, and I've noticed this. I agree that this should be addressed in separate RP

Reply:
I logged an issue for that: https://github.com/ConsenSys/teku/issues/6872